conda_forge_tick/update_prs.py

```diff
@@ -176,14 +177,38 @@ def close_dirty_prs(gx: nx.DiGraph, dry_run: bool = False) -> nx.DiGraph:
     return gx
 
 
-def main(args: "CLIArgs") -> None:
-    # get current time
-    now = datetime.now()
-    current_time = now.strftime("%d-%m-%Y") + "_" + now.strftime("%H_%M_%S")
+# profiling decorator
+def profiling():
+    def _profiling(f):
+        @wraps(f)
+        def __profiling(*rgs, **kwargs):
+            prof = cProfile.Profile()
```

Comment:
You might consider structuring this as a context manager, that way we make certain to clean up even if the underlying function fails

Reply:
I will do that, thanks for the comment

Reply:
just reading more about, I don't have much experience with

Reply:
`cProfile` have a supported Profile class that can be used as an Contex Manager, like this:
```python
with cProfile.Profile() as pr:
    # ... do something ...
```